src/bluesearch/database/topic.py

```diff
@@ -295,3 +297,63 @@ def get_topics_for_pmc_article(
             journal_topics.append(descriptor["name"])
 
     return journal_topics
+
+
+def extract_info_from_zipfile(path: pathlib.Path) -> tuple[str, str]:
+    """Extract topic of a medRxiv/bioRxiv article.
+
+    The `.meca` file should always have a fixed structure. Namely,
+    there is a folder `content` and inside of it there should be
+    a single `.xml` file containing the text and the metadata of the
+    article.
+
+    Parameters
+    ----------
+    path
+        Path to a `.meca` file (which is nothing else
+        than a zip archive) with a fixed structured.
+
+    Returns
+    -------
+    topic : str
+        The subject area of the article.
+    journal : str
+        The journal the article was published in. Should be either
+        "medRxiv" or "bioRxiv".
+
+    Raises
+    ------
+    ValueError
+        Appropriate XML not found or the journal or topic are missing.
+    """
+    with zipfile.ZipFile(path) as myzip:
+        xml_files = [
+            x
+            for x in myzip.namelist()
+            if x.startswith("content/") and x.endswith(".xml")
+        ]
+
+        if len(xml_files) != 1:
+            raise ValueError(
+                "There needs to be exactly one .xml file inside of content/"
+            )
+
+        xml_file = xml_files[0]
+        pattern_topic = """<subj-group subj-group-type="hwp-journal-coll">\r\n<subject>(.*)</subject>"""  # noqa
+        pattern_journal = "<journal-title>(.*)</journal-title>"
+
+        with myzip.open(xml_file, "r") as f:
+            text = f.read().decode("utf-8")
+            match_topic = re.search(pattern_topic, text)
+            match_journal = re.search(pattern_journal, text)
+
+            if match_topic is None:
+                raise ValueError("No topic found")
+
+            if match_journal is None:
+                raise ValueError("No journal found")
+
```

Comment:
Just out of curiosity, did you test this on real data ? Did you encounter the case when only one of the two variables is `None` ? Do you think we should still raise an error in that context ?

Reply:
Yeh, I tested it on real data and I actually never stumbled upon a paper that would have any of the above entries missing. However, maybe it will happen. Under the current setup, this would completely stop the execution of the entrypoint. Do we have some convention in place on what to put inside of the output `.jsonl` in case the topic extraction failed? And should I implement it?

Reply:
I think for pmc we put a `None` to the value when there was a problem related to requests and an empty list when nothing was found but the pipeline went through. However, I am not 100% sure anymore. We can maybe keep it this way for now and adapt it later if needed.